lesson24/script.js

```diff
@@ -0,0 +1,199 @@
+//1.Подсчитать сумму всех чисел в заданном пользователем диапазоне.
+
+let firstNumber = prompt("Введите пожалуйста первое число:");
+let secondNumber = prompt("Введите пожалуйста Второе число:");
+let sum = 0;
+for (i = firstNumber; i <= secondNumber; i++) {
+  sum += firstNumber++;
+}
+alert(`Ваш ответ: ${sum}`);
+
+//2.Запросить 2 числа и найти только наибольший общий делитель.
+
+let firstNumber = prompt("Введите пожалуйста первое число:");
+let secondNumber = prompt("Введите пожалуйста Второе число:");
+let constant = 0;
+
+if (secondNumber > firstNumber) {
+  constant = secondNumber;
+} else {
+  constant = firstNumber;
+}
+for (i = constant; i >= 1; i--) {
+  if (secondNumber % i == 0 && firstNumber % i == 0) {
+    alert(`Наибольший общий делитель: ${i}`);
+    break;
+  }
+}
+
+// document.write(`Наибольший общий делитель: ${i}`);
+//Нашел такой вариант вывода ответа.Думаю в Примерах где ответ слишком большой и занимает много места ето норм сработалоб)
+
+//3.Запросить у пользователя число и вывести все делители этого числа.
+
+let number = prompt("Введите пожалуйста число:");
+console.log(`Делители для числа - ${number}`);
+for (i = number - 1; i >= 1; i--) {
+  if (number % i == 0) {
+    console.log(`${i}`);
+  }
+}
+//С console.log и document.write работает все супер а если alert выводит каждый делитель в новом окне а не все в одном
+
+//4.Определить количество цифр в введенном числе.
+
+let number = prompt("Введите пожалуйста число:");
+let variable = 0;
+do {
+  number /= 10;
+  variable++;
+} while (number >= 1);
+alert(`${variable}`);
+
+//5.Запросить у пользователя 10 чисел и подсчитать, сколько он ввел положительных, отрицательных и нулей. При этом также посчитать, сколько четных и нечетных. Вывести статистику на экран. Учтите, что достаточно одной переменной (не 10) для ввода чисел пользователем.
+
+let zeros = 0;
+let positive = 0;
+let negative = 0;
+let notEven = 0;
+let even = 0;
+let number;
+for (i = 0; i < 10; i++) {
+  number = prompt(
+    "Введите пожалуйста 10 чисел(да ето долго,но я сам чуть мозг не сломал)"
+  );
+  if (number == 0) zeros++;
+  if (number > 0) positive++;
+  if (number < 0) negative++;
+  if (number % 2 != 0 || number != 0) notEven++;
+  if (number % 2 == 0 || number != 0) even++;

```

Comment:
- лучше использовать строгое равенств
- не забывай приводить к числу, если нужно
- 0 - чётное число

```suggestion
  number = +prompt(
    "Введите пожалуйста 10 чисел(да ето долго,но я сам чуть мозг не сломал)"
  );
  if (number === 0) zeros++;
  if (number > 0) positive++;
  if (number < 0) negative++;
  if (number % 2 !== 0) notEven++;
  if (number % 2 === 0) even++;
```